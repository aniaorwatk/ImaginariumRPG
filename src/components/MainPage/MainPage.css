.wrapper {
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    min-width: 100vw;
    min-height: 100vh;
    padding-bottom: 10px;
    background: linear-gradient(0deg, rgba(15, 26, 65, 1) 0%, rgba(9, 19, 52, 0) 100%);
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
}

.wrapper__main-right,
.wrapper__main,
.wrapper__main-right--media {
    display: flex;
    flex-direction: column;
    align-items: center;
}


@media only screen and (min-width: 1439px) {
    .wrapper {
        overflow: hidden;
    }

    .wrapper__main {
        display: flex;
        flex-direction: row-reverse;
    }

    .wrapper__main-right {
        display: flex;
        flex-direction: column-reverse;
    }

    .wrapper__main-right--media {
        display: flex;
        flex-direction: column;
        align-items: center;
    }
}