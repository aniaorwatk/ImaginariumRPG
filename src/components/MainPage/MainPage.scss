@import "./../scss/Main";

.wrapper {
    display: flex;
    position: relative;
    flex-direction: column;
    align-items: center;
    min-width: 100vw;
    min-height: 100vh;
    padding-bottom: 10px;
    background: $color-bg;
    filter: $color-filter-wrapper;

    &__main-right,
    &__main,
    &__main-right--media {
        display: flex;
        flex-direction: column;
        align-items: center;
    }
}

@media only screen and (min-width: 1439px) {
    .wrapper {
        overflow: hidden;

        &__main {
            flex-direction: row-reverse;
            align-items: flex-start;
            justify-content: space-between;
            width: 912px;
        }

        &__main-right {
            flex-direction: column-reverse;
        }

        &__main-right--media {
            flex-direction: column;
            align-items: center;
        }
    }
}
