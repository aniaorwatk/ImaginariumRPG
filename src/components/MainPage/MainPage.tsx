import { useEffect, useState } from "react";
import { getAction, IBannerType, IDataType, IEventType, IMediaType, INewPublicationType, URL_IMAGINARIUM } from "../../API/APIGet";
import Banner from "../Banner/Banner";
import Contact from "../Contact/Contact";
import Events from "../Events/Events";
import Header from "../Header/Header";
import Media from "../Media/Media";
import "./MainPage.scss";

const MainPage = () => {

  const [banner, setBanner] = useState<IBannerType>()
  const [newPublication, setNewPublication] = useState<INewPublicationType>()
  const [events, setEvents] = useState<IEventType[]>([])
  const [patronite, setPatronite] = useState("")
  const [media, setMedia] = useState<IMediaType[]>([])

  useEffect(() => {
    getAction(URL_IMAGINARIUM, (data: IDataType) => {
      setBanner(data.banner)
      setNewPublication(data.new_publication)
      setEvents(data.events)
      setPatronite(data.patronite_email)
      setMedia(data.media)
    })
  }, []);

  return (
    <div className="wrapper">
      <Header
        patronite={patronite}
      />
      <main className="wrapper__main">
        <div className="wrapper__main-right">
          <div className="wrapper__main-right--media">
            <Contact />
            <Media
              media={media}
            />
          </div>
          <Banner
            banner={banner}
          />
        </div>
        <Events
          events={events}
          newPublication={newPublication}
        />
      </main>
    </div>
  );
}

export default MainPage
