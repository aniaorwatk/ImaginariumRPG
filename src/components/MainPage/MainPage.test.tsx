import { screen, render } from "@testing-library/react";
import { unmountComponentAtNode } from "react-dom";
import MainPage from "./MainPage";

describe('MainPage Component', () => {
    let container: any = null;

    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    it('Should render', async () => {
        const { asFragment } = render(<MainPage />, container);

        await screen.findAllByText('Discord');

        expect(asFragment()).toMatchSnapshot();
    })

    it('Should display have two events in Media', async () => {
        render(<MainPage/>, container);

        const event1 = await screen.findAllByText('Discord');
        const event2 = await screen.findAllByText('Twitch');

        expect(event1.length).toBe(1);
        expect(event2.length).toBe(1);
    })

    it('Should display have two events in Publication', async () => {
       
        render(<MainPage/>, container);

        const event1 = await screen.findAllByText(/Title 1/i);
        const event2 = await screen.findAllByText(/Title 2/i);

        expect(event1.length).toBe(1);
        expect(event2.length).toBe(1);
    })

    it('Should render banner', async () => {
        const { asFragment } = render(<MainPage/>);

        await screen.findByAltText(/Baner świąteczny/i);

        expect(asFragment()).toMatchSnapshot();
    })

    it('Should render newPublication', async () => {
        const { asFragment } = render(<MainPage/>);

        await screen.findByText(/Każdy jest Johnem nowa edycja/i);

        expect(asFragment()).toMatchSnapshot();
    })
})
