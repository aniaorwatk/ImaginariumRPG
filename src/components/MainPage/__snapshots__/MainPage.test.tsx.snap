// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Component Should render 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <header
      class="header"
    >
      <div
        class="header__bogumil"
        title="Bogumił in Love"
      />
      <div
        class="header__logo"
      >
        <div
          class="header__logo-bg"
        >
          <img
            alt="Logo Imaginarium"
            height="99"
            loading="eager"
            src="white_logo.svg"
            title="Logo Imaginarium"
            width="103"
          />
        </div>
      </div>
      <h1
        class="header__h1"
      >
        <span
          class="header__title"
        >
          Imaginarium RPG
        </span>
        <span
          class="header__information"
        >
          Sesje RPG | Warsztaty | Bohomazki
        </span>
      </h1>
      <meta
        content="grupa Imaginarium RPG - sesje gier fabularnych Imaginarium "
        name="description"
      />
      <div
        class="patronite"
      >
        <a
          href="https://patronite.pl/imaginariumrpg"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="patronite__logo"
            title="Logo Patronite"
          >
            patronite.svg
          </svg>
        </a>
        |
        <svg
          class="patronite_logoEmail"
        >
          mail.svg
        </svg>
        <a
          class="patronite__email"
          href="mailto:imagrpg@gmail.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          imagrpg@gmail.com
        </a>
      </div>
    </header>
    <main
      class="wrapper__main"
    >
      <div
        class="wrapper__main-right"
      >
        <div
          class="wrapper__main-right--media"
        >
          <div
            class="contact"
          >
            <img
              alt="icon message"
              class="contact__img"
              height="23"
              loading="eager"
              src="fe_messanger.svg"
              title="icon message"
              width="23"
            />
            <p
              class="contact__text"
            >
              kontakt
            </p>
          </div>
          <div
            class="media"
          >
            <a
              class="mediaLink"
              href="www.test1.pl"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="mediaLink__icon"
              >
                <div
                  class="mediaLink__icon-img"
                >
                  <img
                    alt="test1.jpg"
                    class="mediaLink__icon-img--view"
                    height="200"
                    loading="eager"
                    src="test1.jpg"
                    title="test1.jpg"
                    width="200"
                  />
                </div>
                <p
                  class="mediaLink__icon-title"
                >
                  Discord
                </p>
              </div>
            </a>
            <a
              class="mediaLink"
              href="www.test2.pl"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="mediaLink__icon"
              >
                <div
                  class="mediaLink__icon-img"
                >
                  <img
                    alt="test2.jpg"
                    class="mediaLink__icon-img--view"
                    height="200"
                    loading="eager"
                    src="test2.jpg"
                    title="test2.jpg"
                    width="200"
                  />
                </div>
                <p
                  class="mediaLink__icon-title"
                >
                  Twitch
                </p>
              </div>
            </a>
          </div>
        </div>
        <img
          alt="Baner świąteczny"
          class="bannerImg"
          height="100"
          loading="eager"
          src="https://a.storyblok.com/f/188945/342x100/34919fd073/banner.jpg"
          title="Baner świąteczny"
          width="342"
        />
      </div>
      <section
        class="publications"
      >
        <section
          class="newPublication"
        >
          <a
            class="newPublication__link"
            href="www.test1.pl"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="newPublication__link-icon"
            >
              new.svg
            </svg>
            <img
              alt="testowy alt"
              class="newPublication__link-img"
              height="100"
              loading="eager"
              src="test1.jpg"
              title="testowy alt"
              width="433"
            />
            <p
              class="newPublication__link-title"
            >
              “Każdy jest Johnem nowa edycja”
            </p>
          </a>
        </section>
        <svg
          class="publications__bogumil"
          title="Bomumil with ring"
        >
          bLotr.svg
        </svg>
        <div
          class="publications__publishingHouse"
        >
          <svg
            class="publications__publishingHouse-img"
          >
            wydawnictwo.svg
          </svg>
        </div>
        <div
          class="publications__boxWithArrows"
        >
          <img
            alt="arrow right"
            class="publications__boxWithArrows-scroll arrowRight   "
            height="334"
            loading="eager"
            src="arrow_icon.svg"
            title="arrow"
            width="58"
          />
          <div
            class="publications__boxWithArrows-events"
          >
            <div
              class="publication"
            >
              <img
                alt="Title 1"
                class="publication__img"
                height="173"
                loading="eager"
                src="test1.jpg"
                title="Title 1"
                width="157"
              />
              <p
                class="publication__title"
              >
                “Title 1”
              </p>
            </div>
            <div
              class="publication"
            >
              <img
                alt="Title 2"
                class="publication__img"
                height="173"
                loading="eager"
                src="test2.jpg"
                title="Title 2"
                width="157"
              />
              <p
                class="publication__title"
              >
                “Title 2”
              </p>
            </div>
          </div>
        </div>
      </section>
    </main>
  </div>
</DocumentFragment>
`;

exports[`MainPage Component Should render banner 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <header
      class="header"
    >
      <div
        class="header__bogumil"
        title="Bogumił in Love"
      />
      <div
        class="header__logo"
      >
        <div
          class="header__logo-bg"
        >
          <img
            alt="Logo Imaginarium"
            height="99"
            loading="eager"
            src="white_logo.svg"
            title="Logo Imaginarium"
            width="103"
          />
        </div>
      </div>
      <h1
        class="header__h1"
      >
        <span
          class="header__title"
        >
          Imaginarium RPG
        </span>
        <span
          class="header__information"
        >
          Sesje RPG | Warsztaty | Bohomazki
        </span>
      </h1>
      <meta
        content="grupa Imaginarium RPG - sesje gier fabularnych Imaginarium "
        name="description"
      />
      <div
        class="patronite"
      >
        <a
          href="https://patronite.pl/imaginariumrpg"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="patronite__logo"
            title="Logo Patronite"
          >
            patronite.svg
          </svg>
        </a>
        |
        <svg
          class="patronite_logoEmail"
        >
          mail.svg
        </svg>
        <a
          class="patronite__email"
          href="mailto:imagrpg@gmail.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          imagrpg@gmail.com
        </a>
      </div>
    </header>
    <main
      class="wrapper__main"
    >
      <div
        class="wrapper__main-right"
      >
        <div
          class="wrapper__main-right--media"
        >
          <div
            class="contact"
          >
            <img
              alt="icon message"
              class="contact__img"
              height="23"
              loading="eager"
              src="fe_messanger.svg"
              title="icon message"
              width="23"
            />
            <p
              class="contact__text"
            >
              kontakt
            </p>
          </div>
          <div
            class="media"
          >
            <a
              class="mediaLink"
              href="www.test1.pl"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="mediaLink__icon"
              >
                <div
                  class="mediaLink__icon-img"
                >
                  <img
                    alt="test1.jpg"
                    class="mediaLink__icon-img--view"
                    height="200"
                    loading="eager"
                    src="test1.jpg"
                    title="test1.jpg"
                    width="200"
                  />
                </div>
                <p
                  class="mediaLink__icon-title"
                >
                  Discord
                </p>
              </div>
            </a>
            <a
              class="mediaLink"
              href="www.test2.pl"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="mediaLink__icon"
              >
                <div
                  class="mediaLink__icon-img"
                >
                  <img
                    alt="test2.jpg"
                    class="mediaLink__icon-img--view"
                    height="200"
                    loading="eager"
                    src="test2.jpg"
                    title="test2.jpg"
                    width="200"
                  />
                </div>
                <p
                  class="mediaLink__icon-title"
                >
                  Twitch
                </p>
              </div>
            </a>
          </div>
        </div>
        <img
          alt="Baner świąteczny"
          class="bannerImg"
          height="100"
          loading="eager"
          src="https://a.storyblok.com/f/188945/342x100/34919fd073/banner.jpg"
          title="Baner świąteczny"
          width="342"
        />
      </div>
      <section
        class="publications"
      >
        <section
          class="newPublication"
        >
          <a
            class="newPublication__link"
            href="www.test1.pl"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="newPublication__link-icon"
            >
              new.svg
            </svg>
            <img
              alt="testowy alt"
              class="newPublication__link-img"
              height="100"
              loading="eager"
              src="test1.jpg"
              title="testowy alt"
              width="433"
            />
            <p
              class="newPublication__link-title"
            >
              “Każdy jest Johnem nowa edycja”
            </p>
          </a>
        </section>
        <svg
          class="publications__bogumil"
          title="Bomumil with ring"
        >
          bLotr.svg
        </svg>
        <div
          class="publications__publishingHouse"
        >
          <svg
            class="publications__publishingHouse-img"
          >
            wydawnictwo.svg
          </svg>
        </div>
        <div
          class="publications__boxWithArrows"
        >
          <img
            alt="arrow right"
            class="publications__boxWithArrows-scroll arrowRight   "
            height="334"
            loading="eager"
            src="arrow_icon.svg"
            title="arrow"
            width="58"
          />
          <div
            class="publications__boxWithArrows-events"
          >
            <div
              class="publication"
            >
              <img
                alt="Title 1"
                class="publication__img"
                height="173"
                loading="eager"
                src="test1.jpg"
                title="Title 1"
                width="157"
              />
              <p
                class="publication__title"
              >
                “Title 1”
              </p>
            </div>
            <div
              class="publication"
            >
              <img
                alt="Title 2"
                class="publication__img"
                height="173"
                loading="eager"
                src="test2.jpg"
                title="Title 2"
                width="157"
              />
              <p
                class="publication__title"
              >
                “Title 2”
              </p>
            </div>
          </div>
        </div>
      </section>
    </main>
  </div>
</DocumentFragment>
`;

exports[`MainPage Component Should render newPublication 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <header
      class="header"
    >
      <div
        class="header__bogumil"
        title="Bogumił in Love"
      />
      <div
        class="header__logo"
      >
        <div
          class="header__logo-bg"
        >
          <img
            alt="Logo Imaginarium"
            height="99"
            loading="eager"
            src="white_logo.svg"
            title="Logo Imaginarium"
            width="103"
          />
        </div>
      </div>
      <h1
        class="header__h1"
      >
        <span
          class="header__title"
        >
          Imaginarium RPG
        </span>
        <span
          class="header__information"
        >
          Sesje RPG | Warsztaty | Bohomazki
        </span>
      </h1>
      <meta
        content="grupa Imaginarium RPG - sesje gier fabularnych Imaginarium "
        name="description"
      />
      <div
        class="patronite"
      >
        <a
          href="https://patronite.pl/imaginariumrpg"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="patronite__logo"
            title="Logo Patronite"
          >
            patronite.svg
          </svg>
        </a>
        |
        <svg
          class="patronite_logoEmail"
        >
          mail.svg
        </svg>
        <a
          class="patronite__email"
          href="mailto:imagrpg@gmail.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          imagrpg@gmail.com
        </a>
      </div>
    </header>
    <main
      class="wrapper__main"
    >
      <div
        class="wrapper__main-right"
      >
        <div
          class="wrapper__main-right--media"
        >
          <div
            class="contact"
          >
            <img
              alt="icon message"
              class="contact__img"
              height="23"
              loading="eager"
              src="fe_messanger.svg"
              title="icon message"
              width="23"
            />
            <p
              class="contact__text"
            >
              kontakt
            </p>
          </div>
          <div
            class="media"
          >
            <a
              class="mediaLink"
              href="www.test1.pl"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="mediaLink__icon"
              >
                <div
                  class="mediaLink__icon-img"
                >
                  <img
                    alt="test1.jpg"
                    class="mediaLink__icon-img--view"
                    height="200"
                    loading="eager"
                    src="test1.jpg"
                    title="test1.jpg"
                    width="200"
                  />
                </div>
                <p
                  class="mediaLink__icon-title"
                >
                  Discord
                </p>
              </div>
            </a>
            <a
              class="mediaLink"
              href="www.test2.pl"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="mediaLink__icon"
              >
                <div
                  class="mediaLink__icon-img"
                >
                  <img
                    alt="test2.jpg"
                    class="mediaLink__icon-img--view"
                    height="200"
                    loading="eager"
                    src="test2.jpg"
                    title="test2.jpg"
                    width="200"
                  />
                </div>
                <p
                  class="mediaLink__icon-title"
                >
                  Twitch
                </p>
              </div>
            </a>
          </div>
        </div>
        <img
          alt="Baner świąteczny"
          class="bannerImg"
          height="100"
          loading="eager"
          src="https://a.storyblok.com/f/188945/342x100/34919fd073/banner.jpg"
          title="Baner świąteczny"
          width="342"
        />
      </div>
      <section
        class="publications"
      >
        <section
          class="newPublication"
        >
          <a
            class="newPublication__link"
            href="www.test1.pl"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="newPublication__link-icon"
            >
              new.svg
            </svg>
            <img
              alt="testowy alt"
              class="newPublication__link-img"
              height="100"
              loading="eager"
              src="test1.jpg"
              title="testowy alt"
              width="433"
            />
            <p
              class="newPublication__link-title"
            >
              “Każdy jest Johnem nowa edycja”
            </p>
          </a>
        </section>
        <svg
          class="publications__bogumil"
          title="Bomumil with ring"
        >
          bLotr.svg
        </svg>
        <div
          class="publications__publishingHouse"
        >
          <svg
            class="publications__publishingHouse-img"
          >
            wydawnictwo.svg
          </svg>
        </div>
        <div
          class="publications__boxWithArrows"
        >
          <img
            alt="arrow right"
            class="publications__boxWithArrows-scroll arrowRight   "
            height="334"
            loading="eager"
            src="arrow_icon.svg"
            title="arrow"
            width="58"
          />
          <div
            class="publications__boxWithArrows-events"
          >
            <div
              class="publication"
            >
              <img
                alt="Title 1"
                class="publication__img"
                height="173"
                loading="eager"
                src="test1.jpg"
                title="Title 1"
                width="157"
              />
              <p
                class="publication__title"
              >
                “Title 1”
              </p>
            </div>
            <div
              class="publication"
            >
              <img
                alt="Title 2"
                class="publication__img"
                height="173"
                loading="eager"
                src="test2.jpg"
                title="Title 2"
                width="157"
              />
              <p
                class="publication__title"
              >
                “Title 2”
              </p>
            </div>
          </div>
        </div>
      </section>
    </main>
  </div>
</DocumentFragment>
`;
